protocol/x/clob/keeper/liquidations.go

```diff
@@ -711,18 +715,19 @@ func (k Keeper) GetPerpetualPositionToLiquidate(
 ) {
 	// Fetch the subaccount from state.
 	subaccount := k.subaccountsKeeper.GetSubaccount(ctx, subaccountId)
-	subaccountLiquidationInfo := k.GetSubaccountLiquidationInfo(ctx, subaccountId)
 
 	numPositions := len(subaccount.PerpetualPositions)
-	indexOffset := k.GetPseudoRand(ctx).Intn(numPositions)
-
-	for i := 0; i < numPositions; i++ {
-		position := subaccount.PerpetualPositions[(i+indexOffset)%numPositions]
-		// Note that this could run in O(n^2) time. This is fine for now because we have less than a hundred
-		// perpetuals and only liquidate once per subaccount per block. This means that the position with smallest
-		// id will be liquidated first.
-		if !subaccountLiquidationInfo.HasPerpetualBeenLiquidatedForSubaccount(position.PerpetualId) {
-			return position.PerpetualId, nil
+	if numPositions > 0 {
```

Comment:
meganit:
cleaner to do
```
if numPositions == 0
      return error
doLoopOperations()
```

Reply:
will leave it as is since we return error when there is no available perpetual to liquidate (error return below)